pavelib/quality.py

```diff
@@ -756,28 +756,65 @@ def _get_xsscommitlint_count(filename):
         return None
 
 
+def _extract_missing_pii_annotations(filename):
+    """
+    Returns the number of uncovered models from the stdout report of django_find_annotations.
+
+    Arguments:
+        filename: Filename where stdout of django_find_annotations was captured.
+
+    Returns:
+        Number of uncovered models.
+    """
+    uncovered_models = None
+    if os.path.isfile(filename):
+        with open(filename, 'r') as report_file:
+            lines = report_file.readlines()
+            uncovered_regex = re.compile(r'^Coverage found ([\d]+) uncovered')
+            for line in lines:
+                uncovered_match = uncovered_regex.match(line)
+                if uncovered_match:
+                    uncovered_models = int(uncovered_match.groups()[0])
+                    break
+    return uncovered_models
+
+
 @task
 @needs('pavelib.prereqs.install_python_prereqs')
+@cmdopts([
+    ("report-dir=", "r", "Directory in which to put PII reports"),
+])
 @timed
 def run_pii_check(options):  # pylint: disable=unused-argument
     """
     Guarantee that all Django models are PII-annotated.
     """
+    default_report_dir = (Env.REPORT_DIR / "pii_report")
+    report_dir = getattr(options, 'report_dir', default_report_dir)
+    output_file = os.path.join(report_dir, 'pii_check_{}.report')
+    uncovered_model_counts = []
     for env_name, env_settings_file in (("CMS", "cms.envs.test"), ("LMS", "lms.envs.test")):
         try:
             print()
             print("Running {} PII Annotation check and report".format(env_name))
             print("-" * 45)
+            run_output_file = six.text_type(output_file).format(env_name.lower())
             sh(
                 "export DJANGO_SETTINGS_MODULE={}; "
                 "code_annotations django_find_annotations "
-                "--config_file .pii_annotations.yml --report_path pii_report/ "
-                "--lint --report --coverage".format(env_settings_file)
+                "--config_file .pii_annotations.yml --report_path {} "
+                "--lint --report --coverage | tee {}".format(env_settings_file, report_dir, run_output_file)
             )
+            uncovered_model_counts.append(_extract_missing_pii_annotations(run_output_file))
 
         except BuildFailure as error_message:
             fail_quality('pii_check', 'FAILURE: {}'.format(error_message))
 
+    uncovered_count = max(uncovered_model_counts)
```

Comment:
^ I am curious why you are taking the `max` here - would it ever be the case that there are more than one entries in `uncovered_model_counts`?

Reply:
Yes - there should *always* be both an LMS and CMS uncovered model count.

Reply:
OK, in that case, should we be summing these values instead of taking the max?

Reply:
Probably not, since they're almost always going to be duplicates.

Reply:
That was a touch call. Some common models would be in both, so adding them would be misleading. However, some models will be in either/or. I went with the maximum of the two - but I'm unsure if this is best. @bmedx ?

Reply:
I think max is the right way to go, personally.